src/schema/updates/manifest.json

```diff
@@ -61,6 +61,20 @@
                       ]
                     },
                     "uniqueItems": true
+                  },
+                  "developer": {
+                    "properties": {
+                      "url": {
+                        "oneOf": [
```

Comment:
I see that homepage_url does have a slightly different JSONSchema:
https://github.com/mozilla/addons-linter/blob/4b80f2b21a3aebe4b6bf6ca2e47cf5f781ab1c41/src/schema/updates/manifest.json#L87-L98
(and obviously also the same in the imported one)

In particular, the property `"type": "string"` shouldn't matter because it is already part of the imported schema data (and actually I think it was likely not necessary, and probably not advisable, to overwrite that from the schema updates), but I'm wondering if `"format": "ignore"` does serve any actual purpose (I haven't digged enough to collect some more of the historical reasons, but that seems to have been added in the initial PR https://github.com/mozilla/addons-linter/pull/1798).

The test coverage seems to suggest that it may not really matter, but may be worth to double-check that.

Reply:
Yeah, `format: ignore` is used to override the `format` of the `homepagee_url`: https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/manifest.json#78 (which is set to `url`).

I didn't add `format: ignore` to `/developer/url` because `format` isn't defined: https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/manifest.json#296-300

Reply:
I see, that seems to be one more detail that has been overlooked when we added the JSONSchema for the `developer` manifest properties.

I did some more digging into the "historical context" and here is what I was able to gather:
- `format: "url"` has been added along with the initial definition of `homepage_url` as part of [Bug 1244805](https://bugzilla.mozilla.org/show_bug.cgi?id=1244805), but based on what I can see in https://hg.mozilla.org/mozilla-central/rev/184854a1ca0e the addition of the `homepage_url` seems to have been slipped into that patch, but without any clear connection to the bug and patch it is part of
- then after ~1y the definition of the "developer" manifest property has been added in [Bug 1282977](https://bugzilla.mozilla.org/show_bug.cgi?id=1282977) but without a `format: "url"` (and without putting it into the ManifestBase as it should have been, as we learned recently)

The result is that:
- setting an invalid url in homepage_url makes the addon invalid because it does not pass the `format: "url"` validation
- setting an invalid url in developer.url is ignored, and it does silently override a valid homepage url if one was specified, and the invalid url ends up in about:addons (in the addon details view, as a link with the non validated url set as the tag href)

This seems wrong and something we may want to also fix on the Firefox side, which means that we may add the missing `format: "url"` property in a future import.

At this point, I think it may be reasonable to set `format: "ignore"` is the schema updates for `developer.url` now in preparation for the schema import that is going to import that change at some point (e.g. when we will import 96 schema if we proceed with fixing it on the Firefox side, maybe after running a code scan on the already submitted addon to double-check if any already signed addon have a developer.url property that would prevent the addon to install successfully once we will properly validate that property on the Firefox side).

@willdurand wdyt?

Reply:
I agree. I added the `format: ignore`. I filed a bug for FF: https://bugzilla.mozilla.org/show_bug.cgi?id=1739602

This part is still unclear to me but I guess we can discuss it in the bug:

> maybe after running a code scan on the already submitted addon to double-check if any already signed addon have a developer.url property that would prevent the addon to install successfully once we will properly validate that property on the Firefox side

Reply:
@willdurand I think that the `"format": "ignore"` is currently in the wrong place (it should be in the `developer.url` sub-property instead of being set on the `developer` property)

Besides that, I think it may be good to also:
- remove the `"type": "string"` override from `"homepage_url` (it will not make any difference in the imported schemas, but as previously discussed it seems wrong to override the type from the schema updates)
- refresh the imported schema (to avoid the change to the manifest.json schema update to become part of the Firefox 95 schema import, where we didn't actually do any change on this `developer.url` property on the Firefox side, not a big deal but it would avoid confusion and it would be a bit more consistent)

Here is a diff of the changes needed to update both the updates schema and imported schemas as described above (I had a local mozilla-unified close locally, and so I switched that to release 94 and refreshed the schema by re-importing the schemas from that version to be able to get a clean patch with no other changes):

```diff
diff --git a/src/schema/imported/manifest.json b/src/schema/imported/manifest.json
index 785d4620..6a9640e7 100644
--- a/src/schema/imported/manifest.json
+++ b/src/schema/imported/manifest.json
@@ -338,6 +338,7 @@
                     "url": {
                       "type": "string",
                       "preprocess": "localize",
+                      "format": "ignore",
                       "oneOf": [
                         {
                           "format": "url"
diff --git a/src/schema/updates/manifest.json b/src/schema/updates/manifest.json
index 015cdbfa..d53ac422 100644
--- a/src/schema/updates/manifest.json
+++ b/src/schema/updates/manifest.json
@@ -63,9 +63,9 @@
                     "uniqueItems": true
                   },
                   "developer": {
-                    "format": "ignore",
                     "properties": {
                       "url": {
+                        "format": "ignore",
                         "oneOf": [
                           {
                             "format": "url"
@@ -87,7 +87,6 @@
         "properties": {
           "homepage_url": {
             "format": "ignore",
-            "type": "string",
             "oneOf": [
               {
                 "format": "url"
```

Would you mind to apply this patch and update this PR?